runtime/runtime/src/lib.rs

```diff
@@ -2324,19 +2440,111 @@ struct ApplyProcessingReceiptState<'a> {
     local_receipts: VecDeque<Receipt>,
     incoming_receipts: &'a [Receipt],
     delayed_receipts: DelayedReceiptQueueWrapper,
+    pipeline_manager: pipelining::ReceiptPreparationPipeline,
+}
+
+trait MaybeRefReceipt {
+    fn as_ref(&self) -> &Receipt;
 }
 
-impl<'a> ApplyProcessingReceiptState<'a> {
-    /// Obtain the next receipt that should be executed.
-    fn next_local_receipt(&mut self) -> Option<Receipt> {
-        self.local_receipts.pop_front()
+impl MaybeRefReceipt for Receipt {
+    fn as_ref(&self) -> &Receipt {
+        self
     }
 }
 
+impl<'a> MaybeRefReceipt for &'a Receipt {
+    fn as_ref(&self) -> &Receipt {
+        *self
+    }
+}
+
+/// Schedule a one receipt for contract preparation.
+///
+/// The caller should call this method again after the returned number of receipts from `iterator`
+/// are processed.
+fn schedule_contract_preparation<'b, R: MaybeRefReceipt>(
+    pipeline_manager: &mut pipelining::ReceiptPreparationPipeline,
+    state_update: &TrieUpdate,
+    mut iterator: impl Iterator<Item = R>,
+) -> Option<usize> {
+    let scheduled_receipt_offset = iterator.position(|peek| {
+        let peek = peek.as_ref();
+        let account_id = peek.receiver_id();
+        let key = TrieKey::Account { account_id: account_id.clone() };
+        let receiver = get_pure::<Account>(state_update, &key);
+        let Ok(Some(receiver)) = receiver else {
+            // Most likely reason this can happen is because the receipt is for an account that
+            // does not yet exist. This is a routine occurrence as accounts are created by sending
+            // some NEAR to a name that's about to be created.
+            return false;
+        };
```

Comment:
Adding

```
if true {
    return false;
}
```

Here in principle disables the whole pipelining thing, but I still see differing hashes. The only things that get added are the iteration and `get_pure` above.